authentication/auth.go

```diff
@@ -388,6 +515,72 @@ func GenerateTokenFromDefaults(schemaURL string, accountServiceURL string, accou
 	return token, ""
 }
 
+// GenerateTokenFromDefaultsV2 coverts a set of DEFAULT values into a JWT
+func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, accountServiceLogOutURL string, urlValues url.Values) (token string, error string) {
+	launcherSchema, validationError := launcherSchemaFromURL(schemaURL)
+	if validationError != "" {
+		return "", validationError
+	}
+
+	claims := make(map[string]interface{})
+	urlValues["account_service_url"] = []string{accountServiceURL}
+	urlValues["account_service_log_out_url"] = []string{accountServiceLogOutURL}
+
+	claims = generateClaimsV2(urlValues, launcherSchema)
+
+	requiredMetadata, error := GetRequiredMetadata(launcherSchema)
+	if error != "" {
+		return "", fmt.Sprintf("GetRequiredMetadata failed err: %v", error)
+	}
+
+	data := make(map[string]interface{})
+
+	for _, metadata := range requiredMetadata {
+		if metadata.Validator == "boolean" {
+			data[metadata.Name] = getBooleanOrDefault(metadata.Name, urlValues, false)
+
+			continue
+		}
+		data[metadata.Name] = getStringOrDefault(metadata.Name, urlValues, metadata.Default)
+	}
+
+	surveyMetadata := make(map[string]interface{})
+	updatedData := make(map[string]interface{})
+
+	if claims["survey_metadata"] != nil {
+		surveyMetadata = claims["survey_metadata"].(map[string]interface{})
+	}
+	for key, value := range data {
+		updatedData[key] = value
+	}
+
+	initialData := surveyMetadata["data"].(map[string]interface{})
+
+	for key, value := range initialData {
+		updatedData[key] = value
+	}
+
+	surveyMetadata["data"] = updatedData
+	claims["survey_metadata"] = surveyMetadata
```

Comment:
I'm a bit confused as to what's going on here?. Is this not essentially reconstructing the format you already had?.

Reply:
This is because the quicklauncher code follows a different path and loads the token from the default values